plugin/src/main/java/com/playmonumenta/scriptedquests/zones/zone/BaseZone.java

```diff
@@ -0,0 +1,158 @@
+package com.playmonumenta.scriptedquests.zones.zone;
+
+import org.bukkit.Axis;
+import org.bukkit.util.Vector;
+
+import com.playmonumenta.scriptedquests.utils.ZoneUtils;
+
+public class BaseZone {
+	protected Vector mPosition;
+	protected Vector mSize;
+
+	public BaseZone(BaseZone other) {
+		mPosition = other.mPosition.clone();
+		mSize = other.mSize.clone();
+	}
+
+	/*
+	 * pos1 and pos2 are used similar to /fill:
+	 * - Both are inclusive coordinates.
+	 * - The minimum/maximum are determined for you.
+	 */
+	public BaseZone(Vector pos1, Vector pos2) {
+		Vector posMin = Vector.getMinimum(pos1, pos2);
+		Vector posMax = Vector.getMaximum(pos1, pos2);
+
+		mPosition = posMin;
+		mSize = posMax.clone();
+		mSize.subtract(posMin);
+		mSize.add(new Vector(1, 1, 1));
+	}
+
+	/*
+	 * Returns the smallest integer coordinate that is inside the zone on each axis.
+	 */
+	public Vector minCorner() {
+		return mPosition.clone();
+	}
+
+	/*
+	 * Sets the smallest integer coordinate that is inside the zone on each axis.
+	 */
+	public void minCorner(Vector other) {
+		Vector newPosition = mPosition.clone();
+		Vector newSize = mSize.clone();
+
+		newSize.add(mPosition);
+		newSize.subtract(other);
+		newPosition = other.clone();
+
+		mPosition = newPosition;
+		mSize = newSize;
+	}
+
+	/*
+	 * Returns the largest integer coordinate that is inside the zone on each axis.
+	 */
+	public Vector maxCorner() {
+		Vector result = trueMaxCorner();
+		result.subtract(new Vector(1, 1, 1));
+		return result;
+	}
+
+	/*
+	 * Sets the largest integer coordinate that is inside the zone on each axis.
+	 */
+	public void maxCorner(Vector other) {
+		Vector newSize = mSize.clone();
+
+		newSize.add(other);
+		newSize.subtract(maxCorner());
+
+		mSize = newSize;
+	}
+
+	/*
+	 * Returns the smallest coordinate that is outside the zone on each axis.
+	 */
+	public Vector trueMaxCorner() {
+		Vector result = mPosition.clone();
+		result.add(mSize);
+		return result;
+	}
+
+	/*
+	 * Sets the smallest coordinate that is outside the zone on each axis.
+	 */
+	public void trueMaxCorner(Vector other) {
+		Vector newSize = mSize.clone();
+
+		newSize.add(other);
+		newSize.subtract(trueMaxCorner());
+
+		mSize = newSize;
+	}
```

Comment:
The name "true" seems confusing to me. Maybe "maxCornerExclusive()" ? Not a strong opinion, just seems odd.